lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -1199,17 +1199,16 @@ namespace Js
 
     BOOL JavascriptOperators::GetOwnProperty(Var instance, PropertyId propertyId, Var* value, ScriptContext* requestContext)
     {
-        BOOL result;
-        if (TaggedNumber::Is(instance))
-        {
-            result = false;
-        }
-        else
+        if (!TaggedNumber::Is(instance))
         {
             RecyclableObject* object = RecyclableObject::FromVar(instance);
-            result = object && object->GetProperty(object, propertyId, value, NULL, requestContext);
+            if (object)
+            {
+                return Property_Found == object->GetPropertyEx(object, propertyId, value, NULL, requestContext); // return false if instance is typed arrary and GetPropertyEx() returns Property_NotFound_NoProto
```

Comment:
Why does this code has to change?  Seems like the RecyclableObject::GetProperty wrapper will suffice here?